### ISSUE TITLE
[proposal] Support yaml as well as jsonc

### ISSUE THREAD

**Issue by fahhem** [devcontainers/spec#203]
Any interest?

Background:
`jsonc`, aka "JSON with comments", while nicer than plain JSON, is still relatively new and doesn't have full ecosystem support yet. It also has some shortcomings shared with JSON, such as not accepting trailing commas in lists/objects (except on a [per-parser exception](https://github.com/microsoft/vscode/issues/102061) basis).

`yaml` can be limited to a JSON equivalent in terms of data structures it supports, and all (or at least most) of the issues with it were allayed in 1.2+ versions of the spec (such as `on` being parsed as `true` when not quoted). It also reduces the need for as many quotes, commas, open and close braces, and first-class supports comments.

**Comment by Alloneway30** [comment]
I agree on YAML 1.2 yet Im fairly new to contributing. Are you suggesting renaming devcontainers from dev containers to in istead  devcontainers.yaml? 

That's what I get from reading your support, either that or Json being a superset of yaml makes the proposal more appealing. Id be willing to reconfig these settings they are straight forward enough

**Comment by s-h-a-d-o-w** [comment]
Doing this would also be great because JSONC support in devcontainer.json is already being advertised everywhere:
https://code.visualstudio.com/docs/devcontainers/devcontainer-cli#_development-containers
https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/adding-a-dev-container-configuration/introduction-to-dev-containers
https://containers.dev/implementors/spec/#devcontainerjson

Why not make support for the proper `.jsonc` file extension part of the spec, so that one doesn't have to add exceptions to tooling?

Also, it's frustrating that one can't use comments in features. Causes them to fail validation. If this is actually ever done, I hope that it will be applied to _all_ configuration files.
